### ISSUE TITLE
network containing BatchNormalization yields NaN

### ISSUE THREAD

**Issue by santisy** [torch/nn#780]
Hello everyone! 
I encountered a strange problem about BatchNormalization.  At the second time I input the data into the network, it yielded nan. The data I input is the same as the first time. After I removed all the BacthNormalization layers, everything seemed fine. 

Note that for some other use, I first got the parameters of the network by `getParameters` but I did not change it (I have checked by torch.sum(parameters), and I commented relative lines out). However, when I commented the `getParameters`, everything turned normal.

I am wondering if `getParameters` has some side effects on `BacthNormalization`?

Plus, all data and networks are on GPU.

**Comment by fmassa** [comment]
Could you provide a code snippet reproducing the problem ? I think this might be related to https://github.com/fmassa/nn/commit/6a6f5f155edd1e3aa943ddb0f0f91ee79d9d5257

**Comment by santisy** [comment]
@fmassa Hi! thanks for replying! 

I was trying to reproduce the problem in a code snippet, but failed. 
I was writing a control optimization program so I used Lua API to embedded Lua in the simulation program. However, when I extracted the part of lua and simply wrote a c++ program wrapped it which only kept loading a same picture as input, nothing happened. 

Following is the main concerned part of the problem.
When I comment the `qTP = qTarModel:getParameters()`, everything turned normal.
But in the program without simulation environment, everything went fine. 

```lua
  qModel = netWorks.buildNetQ()
  uModel = netWorks.buildNetU()
  qTarModel = qModel:clone()
  uTarModel = uModel:clone()
  parameters, gradParameters = qModel:getParameters()
  uParameters, uGradParameters = uModel:getParameters() 
  uTP = uTarModel:getParameters()
  qTP = qTarModel:getParameters() **<< when commentintg this line, problem disappeared**
```

Every model here is a neural network.
